File: account/phone-numbers/get-company-phone-number-list/code-samples/listAccountPhoneNumbers.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
const accountId = '<ENTER VALUE>';

// OPTIONAL QUERY PARAMETERS
const queryParams = {
    //page: 1,
    //perPage: 100,
    //usageType: [ 'MainCompanyNumber', 'AdditionalCompanyNumber', 'CompanyNumber', 'DirectNumber', 'CompanyFaxNumber', 'ForwardedNumber', 'ForwardedCompanyNumber', 'ContactCenterNumber', 'ConferencingNumber', 'MeetingsNumber' ]
}

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.get(`/restapi/v1.0/account/${accountId}/phone-number`, queryParams).then((r) => {
        // PROCESS RESPONSE
    });
});

File: account/phone-numbers/get-company-phone-number-list/code-samples/listAccountPhoneNumbers.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string accountId = "<ENTER VALUE>";

// OPTIONAL QUERY PARAMETERS
ListAccountPhoneNumbersParameters listAccountPhoneNumbersParameters = new ListAccountPhoneNumbersParameters {
    //page = 1,
    //perPage = 100,
    //usageType = new[] { "MainCompanyNumber", "AdditionalCompanyNumber", "CompanyNumber", "DirectNumber", "CompanyFaxNumber", "ForwardedNumber", "ForwardedCompanyNumber", "ContactCenterNumber", "ConferencingNumber", "MeetingsNumber" }
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Account(accountId).PhoneNumber().List(listAccountPhoneNumbersParameters);
// PROCESS RESPONSE
